python/bifrost/map.py

```diff
@@ -142,3 +145,31 @@ def map(func_string, data, axis_names=None, shape=None,
                      narg, _array(args), _array(arg_names),
                      func_name, func_string, extra_code,
                      _array(block_shape), _array(block_axes)))
+
+def list_cache():
+    output = ''
+    if BF_GPU_MAP_CACHE:
+        cache_path = os.path.join(os.path.expanduser('~'), 'map_cache')
+        try:
+            with open(os.path.join(cache_path, 'cache.version'), 'r') as fh:
+                version = fh.read()
+            runtime, driver = version.split(None, 1)
+            runtime = int(runtime, 10)
+            runtime = "%i.%i" % (runtime//1000, (runtime//10) % 1000)
+            driver = int(driver, 10)
+            driver = "%i.%i" % (driver//1000, (driver//10) % 1000)
+            
+            entries = glob.glob(os.path.join(cache_path, '*.inf'))
+            
+            output += "Cache version: %s (runtime), %s (driver)\n" % (runtime, driver)
+            output += "Cache entries: %i" % len(entries)
+        except OSError:
+            pass
+            
+    if len(output) == 0:
+        output = 'none'
+    print(output)
+
+
+def clear_cache():
+    _check(_bf.bfMapClearCache())
```

Comment:
I also tried building this branch with `--disable-map-cache`, and adjusted the imports and naming so that `map_clear_cache` in the test `setUp` comes here. But then this `_check` fails with `BF_STATUS_UNSUPPORTED` (because map cache is turned off). So we need to decide if it's okay to call clear_cache if cache is disabled... one POV is that such an operation could alway succeed? ("Yep, cache is clear. Because there isn't one!") The alternative, I guess, is to guard against calling it here, or in the `setUp` methods. But fewer conditionals are better than more. :)

Reply:
Yeah, I think always returning `BF_STATUS_SUCCESS` would be better approach if the cache is disabled.  That style is what I used for the cache listing function in `bifrost.map`.